chain/client/src/client.rs

```diff
@@ -147,10 +153,12 @@ pub struct Client {
     /// Approvals for which we do not have the block yet
     pub pending_approvals:
         lru::LruCache<ApprovalInner, HashMap<AccountId, (Approval, ApprovalType)>>,
+    /// A mapping from the first block of an epoch that we know needs to be state synced for some shards
+    /// to a tracker that will find an appropriate sync_hash for state sync
+    catchup_tracker: HashMap<CryptoHash, NewChunkTracker>,
```

Comment:
mini nit: Intuitively it would make sense to track state sync per epoch (epoch_id). Would it make sense in practice?

Reply:
yup, agreed

Reply:
<brainstorm> Perhaps we could even change the sync hash to state sync id that is the epoch id. I'm not sure if it makes sense, how much work would it take and if it's worth it though </brainstorm>